zebra/zebra_vty.c

```diff
@@ -3716,11 +4051,21 @@ void zebra_vty_init(void)
 	install_element(VIEW_NODE, &show_ip_rpf_addr_cmd);
 
 	install_element(CONFIG_NODE, &ipv6_route_blackhole_cmd);
-	install_element(VRF_NODE, &ipv6_route_blackhole_vrf_cmd);
+	if (vrf_is_backend_netns()) {
+		install_element(VRF_NODE,
+				&ipv6_route_blackhole_namespace_vrf_cmd);
+
+		install_element(VRF_NODE,
+		      &ipv6_route_address_interface_namespace_vrf_cmd);
+		install_element(VRF_NODE, &ipv6_route_namespace_vrf_cmd);
+	} else {
```

Comment:
an else is really not needed?  We just need to add the extra namespace defuns when we are using a namespace

Reply:
the implementation issue is that if we don't want to have the same helper but the same command line with some common options, then we have the if else to handle.

Reply:
The else is removed, because of vtysh limitation :  #2232.
so now, there is only one command with many parameters.
 and in the vrf backend with netns case, the table parameter will be used.
in other vases, the parameter will be refused.